### PR TITLE
Realtime output detected protocol

### PR DESCRIPTION
**Pull request by Bit-Warrior-X**

### CODE REVIEW

File: example/ndpiReader.c [hunk 1]
```diff
@@ -2561,6 +2561,121 @@ static void debug_printf(u_int32_t protocol, void *id_struct,
...
+	char stime[30], date[64];
+	struct tm tm;
+	int ret;
+	//char	szquery[512];
+
+	time_t firsttime;
+	firsttime = flow->first_seen_ms / TICK_RESOLUTION;
+	localtime_r(&firsttime, &tm);
+	
+	strftime(stime, sizeof(stime), "%T", &tm );
```
Comment (utoni) [inline example/ndpiReader.c:2640]: Please replace the time format `%T` with `%H:%M:%S`.

Comment (Bit-Warrior-X) [reply example/ndpiReader.c:2640]: Thank you for your advice. I will fix it.

File: example/ndpiReader.c [hunk 2]
```diff
@@ -2561,6 +2561,121 @@ static void debug_printf(u_int32_t protocol, void *id_struct,
...
+	char srcip[64], dstip[64];
+	char app_name[64];
+	char stime[30], date[64];
+	struct tm tm;
+	int ret;
+	//char	szquery[512];
+
+	time_t firsttime;
+	firsttime = flow->first_seen_ms / TICK_RESOLUTION;
+	localtime_r(&firsttime, &tm);
```
Comment (utoni) [inline example/ndpiReader.c]: Please use something like this:

```suggestion
    struct tm result;
#ifdef WIN32
      struct tm * tm_ptr = localtime(&firsttime);
      result = *tm_ptr;
#else
      localtime_r(&firsttime, &result);
#endif
```

**Review by IvanNardi — CHANGES_REQUESTED** [review]
These callbacks might be useful.
However, could we make this logic configurable (via a `ndpiReader` flag), and disabled by default, please?

File: example/ndpiReader.c [hunk 3]
```diff
@@ -2561,6 +2561,127 @@ static void debug_printf(u_int32_t protocol, void *id_struct,
 }
 #endif
 
+/**
+ * @brief From IPPROTO to string NAME
+ */
+static char* ipProto2Name(u_int16_t proto_id) {
```
Comment (IvanNardi) [inline example/ndpiReader.c:2567]: Could you use `ndpi_get_ip_proto_name` instead?

File: example/ndpiReader.c [hunk 4]
```diff
@@ -2561,6 +2561,127 @@ static void debug_printf(u_int32_t protocol, void *id_struct,
...
+  case IPPROTO_IGMP:
+    return("IGMP");
+    break;
+  }
+
+  snprintf(proto, sizeof(proto), "%u", proto_id);
+  return(proto);
+}
+
+static int specific_protos[12] = {
```
Comment (IvanNardi) [inline example/ndpiReader.c:2595]: Could you move this array into `is_specific_protocol()`, please?

Comment (Bit-Warrior-X) [reply example/ndpiReader.c:2595]: Yeah, that's good.

**Comment by IvanNardi** [comment]
@Anton-89, any updates on this PR?

**Comment by utoni** [comment]
Replaced by #2197.

**Pull request closed without merging**